openhcl/virt_mshv_vtl/src/lib.rs

```diff
@@ -615,18 +615,15 @@ impl UhProcessorBox {
         control: Option<&'a mut IdleControl>,
     ) -> Result<UhProcessor<'a, T>, Error> {
         if let Some(control) = &control {
-            let vp_index = self.vp_info.base.vp_index;
+            let vp_index = self.vp_info.base.vp_index.index();
 
             let mut current = Default::default();
             affinity::get_current_thread_affinity(&mut current).unwrap();
-            assert_eq!(&current, CpuSet::new().set(vp_index.index()));
+            assert_eq!(&current, CpuSet::new().set(vp_index));
 
             self.partition
                 .hcl
-                .set_poll_file(
-                    self.partition.vp(vp_index).unwrap().cpu_index,
-                    control.ring_fd().as_raw_fd(),
-                )
+                .set_poll_file(vp_index, control.ring_fd().as_raw_fd())
```

Comment:
The code before used `cpu_index`, which semantically has different meaning from `vp_index` (it is the Linux kernel's CPU number). These appear to always match in practice, but I still want to keep them separate in case this breaks in the future.

Reply:
If we're keeping it we should add some docs on when to use which. Suggestion? I'm guessing we should use cpu_index when talking to non-mshv linux kernel things?

Reply:
Yeah, the rules are to use `cpu_index` when talking to the kernel, `vp_index` when talking to the hypervisor or ourselves (we don't support "remapping" the hypervisor's vp index).